.github/workflows/test_ci.yml

```diff
@@ -50,10 +50,41 @@ jobs:
           path: artifacts
           if-no-files-found: ignore
           retention-days: 5
-      - name: Integration - dappeteer
-        if: ${{ !contains(github.event.head_commit.message, '[skip test]') }}
-        uses: mujo-code/puppeteer-headful@master
+  synpress-e2e:
+    name: Synpress e2e test
+    runs-on: ubuntu-latest
+    steps: 
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+
+      - name: Cache Node Modules
+        uses: actions/cache@v3
         env:
-          CI: "true"
+          cache-name: cache-node-modules
         with:
-          args: npm run integration:headful
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Install Packages
+        run: |
+          git config --global url."https://".insteadOf git://
+          git config --global url."https://".insteadOf git+https://
+          git config --global url."https://".insteadOf ssh://git
+          npm i
+        env:
+          CYPRESS_CACHE_FOLDER: ${{ github.workspace }}/node_modules/.cache/Cypress
```

Comment:
curious what does these do? 🤔

Reply:
the cache node module, caches the node module in the given path, then to use cypress in the github actions we need to declare a cypress cache folder and store the installed cypress there for later use in the e2e test case.

Reply:
then for install packages would `npm i` suffice?

Reply:
yup its gd enough